subprojects/core/src/main/java/org/gradle/execution/DefaultTaskSelector.java

```diff
@@ -94,34 +95,29 @@ private RuntimeException throwTaskSelectionException(SelectionContext context, P
         String searchContext = getSearchContext(targetProject, includeSubprojects);
 
         if (context.getOriginalPath().getPath().equals(taskName)) {
-            throw getProblemsService().getInternalReporter().throwing(builder -> {
-                if (!matcher.getMatches().isEmpty()) {
-                    builder.id("ambiguous", "task matches ambiguous tasks", GradleCoreProblemGroup.taskSelection());
-                } else if (!matcher.getCandidates().isEmpty()) {
-                    builder.id("no-matches", "task has no matches", GradleCoreProblemGroup.taskSelection());
-                } else {
-                    builder.id("failed", "task selection failed", GradleCoreProblemGroup.taskSelection());
-                }
-
+            throw getProblemsService().getInternalReporter().throwing(spec -> {
+                configureProblem(spec, matcher, context);
                 String message = matcher.formatErrorMessage("Task", searchContext);
```

Comment:
Now looking at [DefaultBuildTaskSelector.java](https://github.com/gradle/gradle/pull/29050/files#diff-65e6695deb07522bc7cab2b82692c989e30c8976c53a71d7b7f4716445a90d77), I'm not certain this would be possible.